sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -139,22 +143,99 @@ public boolean existsById(Long id, DbSession session) {
     return mapper(session).selectComponentsHavingSameKeyOrderedById(key);
   }
 
-  public List<ComponentDtoWithSnapshotId> selectDirectChildren(DbSession dbSession, ComponentTreeQuery componentQuery) {
-    RowBounds rowBounds = new RowBounds(offset(componentQuery.getPage(), componentQuery.getPageSize()), componentQuery.getPageSize());
-    return mapper(dbSession).selectDirectChildren(componentQuery, rowBounds);
+  /**
+   * Optional parent. It is absent if specified component is root.
+   */
+  public Optional<ComponentDto> selectParent(DbSession dbSession, ComponentDto component) {
```

Comment:
weren't we suppose to move to a `String uuid` argument rather than a `ComponentDto component` ?